### ISSUE TITLE
networkx harmonic centrality vs. closeness centrality

### ISSUE THREAD

**Issue by michaelkkim** [networkx/networkx#2238]
I'm currently running networkx 1.11

**Issue:**
When I run networkx.harmonic_centrality(G), there is a huge memory leakage (ran on ~60,000 nodes). I checked the documentation page, and it's only listed under 'latest' version but not under 'stable' or 'networkx-1.11' versions. Even though I checked that this function does exist in networkx 1.11, is it an unstable function currently? 

If so, how can I tweak parameters/source codes to run harmonic centrality on all my nodes as described in 'latest' version. 
             **i.e.** Would using non-normalized networkx.closeness_centrality(G) be the same as              network.harmonic_centrality(G)?
              **i.e.** Would dividing all my closeness_centrality values by n-1 be the same as networkx.harmonic_centrality(G)?

Thank you.

**Comment by hagberg** [comment]
Harmonic centrality, sum(1/distance),  and closeness centrality, 1/sum(distance) are different.

Can you be more specific about what you mean by memory leak?   The code for harmonic centrality is pretty simple:
```python
from __future__ import division
import networkx as nx
def harmonic_centrality(G, distance=None):
    if G.is_directed():
        G = G.reverse()
    sp = nx.shortest_path_length(G, weight=distance)
    return {n: sum(1 / d if d > 0 else 0 for d in dd.values()) for n, dd in sp}
```

**Comment by michaelkkim** [comment]
It may not exactly be a memory leak, but when running harmonic_centrality, I noticed that it was taking up a considerable amount of memory (3gb ish). It might not be networkx's fault, but I was able to compute other centrality types with much less memory intake.

**Comment by hagberg** [comment]
Oh, I think I see the issue.  The closeness centrality code compute single source shortest path in a loop over every source which takes much less memory than the harmonic centrality implementation that uses all-pairs shortest path.  I think the latter could be easily fixed to be more memory efficient.

**Comment by michaelkkim** [comment]
Ah, I should've tried those functions myself. Thank you for pointing that out.

**Comment by jfinkels** [comment]
@hagberg Do you want the single-source shortest path version in NetworkX?

**Comment by hagberg** [comment]
Yes. That should be very similar in runtime and use much less memory.

**Comment by hagberg** [comment]
Fixed in #2247.
